### PR TITLE
[Banner] Add new feature including config, yang and tests

### PR DESCRIPTION
**Pull request by SviatoslavBoichuk**
#### Why I did it
Added Banner feature related services according to HLD: https://github.com/sonic-net/SONiC/pull/1361

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Added banner-config systemd service, YANG model for new ConfDB table and YANG model tests

#### How to verify it
Manual test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

### CODE REVIEW

File: files/image_config/bannerconfig/banner-config.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+if [[ $STATE == "enabled" ]]; then
+    LOGIN=$(sonic-cfggen -d -v 'BANNER_MESSAGE["MESSAGE"]["login"]')
+    MOTD=$(sonic-cfggen -d -v 'BANNER_MESSAGE["MESSAGE"]["motd"]')
+    LOGOUT=$(sonic-cfggen -d -v 'BANNER_MESSAGE["MESSAGE"]["logout"]')
+
+    echo -e "$LOGIN" > /etc/issue.net
+    echo -e "$LOGIN" > /etc/issue
+    echo -e "$MOTD" > /etc/motd
+    echo -e "$LOGOUT" > /etc/logout_message
+fi
```
Comment (sanal-siva) [inline files/image_config/bannerconfig/banner-config.sh:17]: If state is disabled, are you planning clear banner files?

Comment (SviatoslavBoichuk) [reply files/image_config/bannerconfig/banner-config.sh:17]: No, we do not clean banner messages.

**Comment by qiluo-msft** [comment]
Please resolve the conflict.

File: src/sonic-yang-models/yang-models/sonic-banner.yang [hunk 2]
```diff
@@ -0,0 +1,50 @@
+module sonic-banner {
+    yang-version 1.1;
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-banner.yang:2]: @ganglyu @wen587 Please review the yang change.

File: files/image_config/bannerconfig/banner-config.service [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Update banner config based on configdb
```
Comment (qiluo-msft) [inline files/image_config/bannerconfig/banner-config.service:2]: I have a later comment on HLD https://github.com/sonic-net/SONiC/pull/1361/files#r1451662677

File: src/sonic-yang-models/yang-models/sonic-banner.yang [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    description "BANNER_MESSAGE YANG Module for SONiC-based OS";
+    revision 2023-05-18 {
+        description "First Revision";
+    }
+    container sonic-banner {
+        container BANNER_MESSAGE {
+            description "BANNER_MESSAGE part of config_db.json";
+            container MESSAGE {
+                leaf state {
+                    type string {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-banner.yang:14]: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L127
Can you use admin_mode from sonic-types.yang?

File: src/sonic-yang-models/yang-models/sonic-banner.yang [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                             ";
+                }
+                leaf logout {
+                    type string;
+                    description "Banner message dispalyed to the users on logout";
+                    default "";
+                }
+            } /* end of container MESSAGE */
+        }
+        /* end of container BANNER_MESSAGE */
```
Comment (wen587) [inline src/sonic-yang-models/yang-models/sonic-banner.yang:47]: please keep the same ending style like line 45

**Comment by SviatoslavBoichuk** [comment]
> Please resolve the conflict.

Fixed conflicts

File: src/sonic-yang-models/yang-models/sonic-banner.yang (hunk 4 above)
Comment (SviatoslavBoichuk) [reply src/sonic-yang-models/yang-models/sonic-banner.yang]: Done

File: src/sonic-yang-models/yang-models/sonic-banner.yang (hunk 5 above)
Comment (SviatoslavBoichuk) [reply src/sonic-yang-models/yang-models/sonic-banner.yang]: Done

**Comment by zhangyanzhao** [comment]
Reviewers, can you please help to check the updates to see if you are ok to approve this PR? Thanks.

**Comment by qiluo-msft** [comment]
@ganglyu @wen587 Do you have more concerns?

**Review by ganglyu — APPROVED** [review]

**Review by wen587 — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by wen587** [comment]
Hi @SviatoslavBoichuk , can we revert the change in init_cfg.json and make it backward compatible?
In Azure internal devices, we want to keep the default behavior when we don't have BANNER_MESSAGE.

**Pull request merged**